bin/include/functions.sh

```diff
@@ -54,22 +76,20 @@ checkJava() {
     #
     # Check JDK.
     #
-    if [ ! -e "$JAVA" ]; then
-        echo $0", ERROR:"
-        echo "JAVA is not found in JAVA_HOME=$JAVA_HOME."
-        echo "Please point JAVA_HOME variable to installation of JDK 1.8 or JDK 9"
-        echo "You can also download latest JDK at http://java.com/download"
+    javaMajorVersion "$JAVA"
 
-        exit 1
-    fi
-
-    "$JAVA" -version 2>&1 | grep -qE 'version "(1.8.*|9.*)"' || {
+    if [ $version -lt 8 ]; then
         echo "$0, ERROR:"
-        echo "The version of JAVA installed in JAVA_HOME=$JAVA_HOME is incorrect."
+        echo "The version of JAVA installed in JAVA_HOME=$JAVA_HOME is incompatible."
         echo "Please point JAVA_HOME variable to installation of JDK 1.8 or JDK 9."
         echo "You can also download latest JDK at http://java.com/download"
         exit 1
-    }
+    elif [ $version -gt 9 ]; then
+        echo "$0, WARNING:"
+        echo "The version of JAVA installed in JAVA_HOME=$JAVA_HOME was not tested with Apache Ignite."
```

Comment:
I would add detected JAVA version (the fuller the better) to output to help user to understand on which JAVA currently Apache Ignite is trying to start